internal/tempfile/tempfile_linux.go

```diff
@@ -20,7 +22,7 @@ func Create(dir string) (*os.File, error) {
 	}
 
 	if errors.Is(err, syscall.EISDIR) || errors.Is(err, syscall.EOPNOTSUPP) {
-		return createUnixFallback(dir)
+		return createUnixFallback()
```

Comment:
This branch is not exercised on CI because the Linux runners have recent enough kernels to succeed in the `O_TMPFILE` above.